src/histogram_equalization.jl

```diff
@@ -127,6 +132,18 @@ function get_rgb_channels(img)
     return cat(redc, greenc, bluec; dims=3)
 end
 
+"""
+    rgb2gray(rgbchannels::Array{Float64, 3})
+
+Convert an array of RGB channel data to grayscale in the range [0, 255].
+"""
+function rgb2gray(rgbchannels::Array{Float64,3})
+    r, g, b = [to_uint8(rgbchannels[:, :, i]) for i in 1:3]
+    # Reusing the r array to store the equalized gray image
+    r .= to_uint8(0.2989 * r .+ 0.5870 * g .+ 0.1140 * b)
```

Comment:
Where do these specific numbers come from? 0.2989, 0.5870, 0.1140

Reply:
That's how the MATLAB's `rgb2gray` converts a RGB image to grayscale (uint8). See [here](https://tinyurl.com/mvceu5h8).

Reply:
Let's add a comment to say the source - "Converts RGB to grayscale using a linear combination of channel brightness values. Identical to MATLAB "rgb2gray" (source link)."

Reply:
I added that to the docstring.